### PR TITLE
Motion magic

### PR DESCRIPTION
**Pull request by tristanduerk**
Cleaned up the code. Added two functions, one that converted feet to ticks and the other is a motion magic turn function.

### CODE REVIEW

File: src/org/usfirst/frc/team467/robot/Drive.java
Comment (npetrangelo) [inline src/org/usfirst/frc/team467/robot/Drive.java]: Should move this to RobotMap.

Comment (npetrangelo) [inline src/org/usfirst/frc/team467/robot/Drive.java]: Should move to RobotMap

File: src/org/usfirst/frc/team467/robot/Drive.java
Comment (npetrangelo) [inline src/org/usfirst/frc/team467/robot/Drive.java]: Instead of commenting what it is, actually write in the math, and comment what it represents.

File: src/org/usfirst/frc/team467/robot/Drive.java
Comment (npetrangelo) [inline src/org/usfirst/frc/team467/robot/Drive.java]: Final code should not reference the SmartDashboard.

File: src/org/usfirst/frc/team467/robot/Drive.java
Comment (npetrangelo) [inline src/org/usfirst/frc/team467/robot/Drive.java]: Same as before, final code should not reference the SmartDashboard.

File: src/org/usfirst/frc/team467/robot/Drive.java
Comment (npetrangelo) [inline src/org/usfirst/frc/team467/robot/Drive.java]: Fix the name of the function. publishRawSensorValues()

File: src/org/usfirst/frc/team467/robot/Drive.java
Comment (npetrangelo) [inline src/org/usfirst/frc/team467/robot/Drive.java]: Add the @Override back in

File: src/org/usfirst/frc/team467/robot/Drive.java
Comment (npetrangelo) [inline src/org/usfirst/frc/team467/robot/Drive.java]: Comment that this is here to fix measurement error.

File: src/org/usfirst/frc/team467/robot/DriveMode.java
Comment (npetrangelo) [inline src/org/usfirst/frc/team467/robot/DriveMode.java]: DriveModes should be arcadeDrive and tankDrive.

File: src/org/usfirst/frc/team467/robot/Robot.java
Comment (npetrangelo) [inline src/org/usfirst/frc/team467/robot/Robot.java:95]: Keep this in so we can deal with deprecated features properly later.

File: src/org/usfirst/frc/team467/robot/Robot.java
Comment (npetrangelo) [inline src/org/usfirst/frc/team467/robot/Robot.java]: The specific values we used don't matter anymore, this from autonomousPeriodic(). In fact, this type of thing should be in testPeriodic() in the future.

File: src/org/usfirst/frc/team467/robot/Robot.java
Comment (npetrangelo) [inline src/org/usfirst/frc/team467/robot/Robot.java]: Keep this stuff in.

**Review by npetrangelo — CHANGES_REQUESTED** [review]
Overall, Motion Magic works great, however you still need to clean up before merging to master.

File: src/org/usfirst/frc/team467/robot/Drive.java
Comment (john-petrangelo) [inline src/org/usfirst/frc/team467/robot/Drive.java]: I don't understand why this code is here instead of inside initMotor(). Generally you want to keep constructors very simple and call other initXYZ() methods to do the actual work. Or maybe you need to an initLeadMotor() method if it needs to be different.

Comment (john-petrangelo) [reply src/org/usfirst/frc/team467/robot/Drive.java]: Also, you didn't write it this time, but it's probably time to remove line 45. It's just a TODO comment that has been marked as Done instead of removed.

Comment (john-petrangelo) [reply src/org/usfirst/frc/team467/robot/Drive.java]: And it is a constant that belongs in the RobotMap.

File: src/org/usfirst/frc/team467/robot/Drive.java
Comment (john-petrangelo) [inline src/org/usfirst/frc/team467/robot/Drive.java]: Same comments as above about the left motors. Also, don't leave in the commented out code -- remove it please.

File: src/org/usfirst/frc/team467/robot/Drive.java
Comment (john-petrangelo) [inline src/org/usfirst/frc/team467/robot/Drive.java:80]: Please un-add these three blank lines.

File: src/org/usfirst/frc/team467/robot/Drive.java
Comment (john-petrangelo) [inline src/org/usfirst/frc/team467/robot/Drive.java:101]: Just remove the TODO comment. You won't forget to update this, it won't compile if the constructor changes.

File: src/org/usfirst/frc/team467/robot/Drive.java
Comment (john-petrangelo) [inline src/org/usfirst/frc/team467/robot/Drive.java:116]: This is good cleanup.

File: src/org/usfirst/frc/team467/robot/Drive.java
Comment (john-petrangelo) [inline src/org/usfirst/frc/team467/robot/Drive.java]: Remove the TODO comment on line 125.

File: src/org/usfirst/frc/team467/robot/Drive.java
Comment (john-petrangelo) [reply src/org/usfirst/frc/team467/robot/Drive.java]: The default values should come from RobotMap.

File: src/org/usfirst/frc/team467/robot/Drive.java
Comment (john-petrangelo) [reply src/org/usfirst/frc/team467/robot/Drive.java]: Same as before, default values should come from RobotMap.

File: src/org/usfirst/frc/team467/robot/Drive.java
Comment (john-petrangelo) [inline src/org/usfirst/frc/team467/robot/Drive.java]: Too many hard-coded numbers. Generally these belong in RobotMap, with very clearly written comments.
Also, why are these all divide by two? (Don't answer me here, be sure that you have comments explaining it in the code.

File: src/org/usfirst/frc/team467/robot/Drive.java
Comment (john-petrangelo) [inline src/org/usfirst/frc/team467/robot/Drive.java:206]: Remove the TODO comment.

File: src/org/usfirst/frc/team467/robot/Drive.java
Comment (john-petrangelo) [inline src/org/usfirst/frc/team467/robot/Drive.java:232]: Delete this TODO comment.

File: src/org/usfirst/frc/team467/robot/Drive.java
Comment (john-petrangelo) [inline src/org/usfirst/frc/team467/robot/Drive.java]: What is the default control mode? In other words, what value does this have before it is assigned?

File: src/org/usfirst/frc/team467/robot/Drive.java
Comment (john-petrangelo) [inline src/org/usfirst/frc/team467/robot/Drive.java]: Just remove the entire method, it's not used.

File: src/org/usfirst/frc/team467/robot/Drive.java
Comment (john-petrangelo) [inline src/org/usfirst/frc/team467/robot/Drive.java]: I think it's safe to remove this unused method.

File: src/org/usfirst/frc/team467/robot/Drive.java
Comment (john-petrangelo) [inline src/org/usfirst/frc/team467/robot/Drive.java]: Remove this function.

File: src/org/usfirst/frc/team467/robot/Drive.java
Comment (john-petrangelo) [reply src/org/usfirst/frc/team467/robot/Drive.java]: Wouldn't it make more sense to just adjust your diameterInInches value to 23.2 inches? This 367.5/360 is just a crude hack, isn't it?

Also, these numbers all belong in RobotMap, not in the Drive class.

File: src/org/usfirst/frc/team467/robot/Drive.java
Comment (john-petrangelo) [inline src/org/usfirst/frc/team467/robot/Drive.java:319]: I recommend doing all the distance math in inches instead of feet. Decimal fractional feed is not natural to think in, IMO.

File: src/org/usfirst/frc/team467/robot/DriveMode.java
Comment (john-petrangelo) [inline src/org/usfirst/frc/team467/robot/DriveMode.java]: Fix the indent.

File: src/org/usfirst/frc/team467/robot/DriverStation.java
Comment (john-petrangelo) [inline src/org/usfirst/frc/team467/robot/DriverStation.java]: I don't understand what this TODO is telling me needs to be done. Please add more explanation.

File: src/org/usfirst/frc/team467/robot/DriverStation.java
Comment (john-petrangelo) [inline src/org/usfirst/frc/team467/robot/DriverStation.java]: What is this method used for? Can it be removed?

File: src/org/usfirst/frc/team467/robot/Robot.java
Comment (john-petrangelo) [reply src/org/usfirst/frc/team467/robot/Robot.java:95]: Usually I say to take out obsolete code, but for the autonomous code we know we're going to start writing that again too, so it does make sense to leave it in this time.

File: src/org/usfirst/frc/team467/robot/Robot.java
Comment (john-petrangelo) [inline src/org/usfirst/frc/team467/robot/Robot.java]: Delete commented out code.

File: src/org/usfirst/frc/team467/robot/Drive.java
Comment (npetrangelo) [inline src/org/usfirst/frc/team467/robot/Drive.java:50]: Better.

File: src/org/usfirst/frc/team467/robot/Robot.java
Comment (john-petrangelo) [reply src/org/usfirst/frc/team467/robot/Robot.java]: I agree, test mode would be better for this sort of thing.

I don't think you want to keep this code. In fact, this is your test with degrees code. If you want to keep this code you probably want a "testWithDegrees()" or similarly named method. It would be nice to be able to switch back and forth from the angle to the distance tests.

File: src/org/usfirst/frc/team467/robot/Robot.java
Comment (npetrangelo) [inline src/org/usfirst/frc/team467/robot/Robot.java]: Fix the whitespace.

File: src/org/usfirst/frc/team467/robot/Robot.java
Comment (npetrangelo) [inline src/org/usfirst/frc/team467/robot/Robot.java:168]: Looks good, but we should test to make sure we didn't break stuff.

**Review by npetrangelo — CHANGES_REQUESTED** [review]
Much better overall, but we should test teleopPeriodic() to make sure we didn't break it. Also, what happened to turbo and slow mode?

File: src/org/usfirst/frc/team467/robot/Robot.java
Comment (john-petrangelo) [inline src/org/usfirst/frc/team467/robot/Robot.java]: I'm not sure this switch is correct. Have you tested all cases?

**Review by john-petrangelo — CHANGES_REQUESTED** [review]

File: src/org/usfirst/frc/team467/robot/Robot.java
Comment (npetrangelo) [inline src/org/usfirst/frc/team467/robot/Robot.java]: Good

File: src/org/usfirst/frc/team467/robot/Robot.java
Comment (npetrangelo) [inline src/org/usfirst/frc/team467/robot/Robot.java]: Crazy amounts of whitespace, what's going on here?

File: src/org/usfirst/frc/team467/robot/Robot.java
Comment (npetrangelo) [inline src/org/usfirst/frc/team467/robot/Robot.java]: Remove this stuff; it was just here for testing purposes.

File: src/org/usfirst/frc/team467/robot/Robot.java
Comment (npetrangelo) [inline src/org/usfirst/frc/team467/robot/Robot.java]: Comment this out too. When we eventually start working on Autonomous, we won't be using this in here.

File: src/org/usfirst/frc/team467/robot/Robot.java
Comment (npetrangelo) [inline src/org/usfirst/frc/team467/robot/Robot.java]: Fix this whitespace. It's tabbed over too much.

**Review by npetrangelo — CHANGES_REQUESTED** [review]
The only things you should change are autonomous testing code and some whitespace. Other than that we should test teleop and make sure we didn't break that.

File: src/org/usfirst/frc/team467/robot/Robot.java
Comment (npetrangelo) [inline src/org/usfirst/frc/team467/robot/Robot.java]: Much better.

File: src/org/usfirst/frc/team467/robot/Robot.java
Comment (npetrangelo) [reply src/org/usfirst/frc/team467/robot/Robot.java]: Now it's fixed. Good.

File: src/org/usfirst/frc/team467/robot/Robot.java
Comment (npetrangelo) [inline src/org/usfirst/frc/team467/robot/Robot.java:153]: Much better now.

**Review by npetrangelo — APPROVED** [review]
Looks good. But wait for another review before merging.

File: src/org/usfirst/frc/team467/robot/Drive.java
Comment (john-petrangelo) [inline src/org/usfirst/frc/team467/robot/Drive.java]: These other numbers belong in the RobotMap, too, not just the timeout value.

File: src/org/usfirst/frc/team467/robot/Drive.java
Comment (john-petrangelo) [reply src/org/usfirst/frc/team467/robot/Drive.java]: Same as above, these numbers belong in RobotMap. Also please remove the commented out setInverted() calls.

File: src/org/usfirst/frc/team467/robot/Drive.java
Comment (john-petrangelo) [inline src/org/usfirst/frc/team467/robot/Drive.java:145]: Why is this all commented out now?

File: src/org/usfirst/frc/team467/robot/Drive.java
Comment (john-petrangelo) [inline src/org/usfirst/frc/team467/robot/Drive.java]: Get the default values from RobotMap. Pretty soon you're going to have more than one robot to configure and you'll need these to vary depending on which robot you are running on.

File: src/org/usfirst/frc/team467/robot/Drive.java
Comment (john-petrangelo) [reply src/org/usfirst/frc/team467/robot/Drive.java]: Still need to put these into RobotMap.

File: src/org/usfirst/frc/team467/robot/Robot.java
Comment (john-petrangelo) [inline src/org/usfirst/frc/team467/robot/Robot.java]: I don't think you want this left on all the time. Maybe drop it down to a trace level message.

**Review by john-petrangelo — CHANGES_REQUESTED** [review]
This looks a lot better, thanks for the cleanup. The biggest thing left is to continue moving constants into RobotMap.

**Review by john-petrangelo — APPROVED** [review]

**Pull request merged**
